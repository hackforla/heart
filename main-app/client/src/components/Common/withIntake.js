import React, { Component } from 'react'

const styles = {
  topHeader: {
    background: '#6f7ce2',
    height: '400px',
  },
}

const withIntake = WrappedComponent => {
  return class extends Component {
    render() {
      return (
        <div className="intake">
          <div style={styles.topHeader}></div>
          <WrappedComponent />
        </div>
      )
    }
  }
}

export default withIntake
