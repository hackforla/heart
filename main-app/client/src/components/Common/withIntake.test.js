import React from 'react';
import renderer from 'react-test-renderer';
import withIntake from './withIntake';

it('renders correctly intake component', () => {
  const WrappedComponent = withIntake(() => <div>I'm wrapped</div>);
  const IntakeComponent = renderer.create(<WrappedComponent />).toJSON();
  expect(IntakeComponent).toMatchSnapshot();
});
