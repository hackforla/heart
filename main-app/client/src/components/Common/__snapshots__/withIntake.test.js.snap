// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly intake component 1`] = `
<div
  className="intake"
>
  <div
    style={
      Object {
        "background": "#6f7ce2",
        "height": "400px",
      }
    }
  />
  <div>
    I'm wrapped
  </div>
</div>
`;
